homeassistant/components/openuv/__init__.py

```diff
@@ -47,13 +53,85 @@
 
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.SENSOR]
 
+SERVICE_NAME_UPDATE_DATA = "update_data"
+SERVICE_NAME_UPDATE_PROTECTION_DATA = "update_protection_data"
+SERVICE_NAME_UPDATE_UV_INDEX_DATA = "update_uv_index_data"
+
+SERVICES = (
+    SERVICE_NAME_UPDATE_DATA,
+    SERVICE_NAME_UPDATE_PROTECTION_DATA,
+    SERVICE_NAME_UPDATE_UV_INDEX_DATA,
+)
+
+SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE_ID): cv.string,
+    }
+)
+
+
+@callback
+def async_get_openuv_for_service_call(hass: HomeAssistant, call: ServiceCall) -> OpenUV:
+    """Get the OpenUV object related to a service call (by device ID)."""
+    device_id = call.data[CONF_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if (device_entry := device_registry.async_get(device_id)) is None:
+        raise ValueError(f"Invalid OpenUV service ID: {device_id}")
```

Comment:
Is this string correct?

Reply:
Roger: this device is of type `DeviceEntryType.SERVICE` (https://github.com/home-assistant/core/blob/e4490145c7f2b77f05389148b89a9710c8fba386/homeassistant/components/openuv/__init__.py#L348).

Reply:
It might be a bit confusing since the UI shows `Choose device`

Reply:
Ooh, you know what... we now have config entry selectors. That would make more sense here. I'll update.